include/envoy/server/options.h

```diff
@@ -11,6 +11,31 @@
 namespace Server {
 
 /**
+ * Whether to run Envoy in serving mode, or in config-validation mode at one of two levels (in which
+ * case we'll verify the configuration file is valid, print any errors, and exit without serving.)
+ */
+enum class Mode {
+  /**
+   * Default mode: Regular Envoy serving process. Configs are validated in the normal course of
+   * initialization, but if all is well we proceed to serve traffic.
+   */
+  Serve,
```

Comment:
It's fine in the enum here, but for the command-line, I'm in agreement with @PiotrSikora, it's probably cleaner to have a `--dry-run` and `--dry-run-lite` flags, I don't think anyone will ever set `--mode=serve`.